### ISSUE TITLE
Investigate how text-annotations work in Neatline and compare notes with the ppl working on text annotation at UWS

### ISSUE THREAD

**Issue by ptsefton** [ozmeka/ozmeka#7]
Discuss with @ptsefton and @TheKingOfThePotatoPeople

**Comment by kevinpdavies** [comment]
Here is my very preliminary understanding of how Neatline annotations work. And some initial discussion on how it might be done better (I'm new to Omeka/Neatline so I might be way off on this).

After an Exhibition is created, an item can then be imported. This item is then given a "slug", as well as other "annotation" data such as map symbol and a map layer residing on a Web Map Service (eg Geoserver WMS). 

At the exhbition level, the narrative text is entered. Specific words can be tagged with the "slug" for an item associated with the exhibit. E.g: this example associates the slug "MDBCatchments" with a place name in the text.

```
Moving upstream, the stories start in South Australia: the <span data-neatline-slug="MDBCatchments">Coorong and Lower Lakes (2)</span>.
```

When the user hovers over the tagged text, it is highlighted on the map, and when they click on it, the map is focussed on the item, and a bubble displayed with the metadata record. This bubble is customisable.

Exhibits are stored in a separate DB table (omeka_neatline_exhibits) which contains exhibit level attributes including the marked-up narrative. The 1-M relationship of exhibits is stored in the omeka_neatline_records table. The exhibit-specific attributes of an item, such as the WMS layer and any map features (stored as a 'geometry' field in mysql) are kept in this table. The original item metadata is not affected. An item could therefore have different spatial attributes for different exhibits.

As we discussed, spatial information should be associated with the item, and an exhibit could override this. However, there is no strict format for 'coverage' in DC as far as I understand. It could just be a place name or a feature (i.e. a river). 

It would be really nice if we could just store space and time descriptors in DC somehow (e.g. Well-Known-Text or GML/KML). I don't know DC that well and whether this is 'allowed'. Such space/time info would need to be enforced at the item entry stage (i.e. mapping tool or geonames lookup). This item level space/time data would could then be integrated into the Neatline mapping plugin and Solr search.

**Comment by kevinpdavies** [comment]
The DC coverage does have support for point and rectangular extent of an item:
    http://dublincore.org/documents/usageguide/elements.shtml#coverage
and includes support for datum/projection. So this is promising.

However, this (old?) working draft also included polygons which would be nice:
    http://dublincore.org/documents/coverage-element/

**Comment by sharynys** [comment]
yep DC.coverage.spatial supports DCMI point (point coords) and DCMI box (polygon coords). Can include projection (didnt know that!) 
http://dublincore.org/documents/dcmi-box/
http://dublincore.org/documents/dcmi-point/

**Comment by ptsefton** [comment]
We're not limited to DC - is there a geo metadata standard we should be supporting? Could add that as a new element set.

**Comment by sharynys** [comment]
We aren't - except in so far as OAI-PMH will require it

**Comment by sharynys** [comment]
Another factor to consider is that a lot of government agencies/funders require ANZLIC metadata in their Open Data Obligations clauses of research contracts. It would therefore be wise to support that (although there is not a great deal of evidence that govt agencies enforce this ...or even know what it is! OEH is on to it, others less so). The contracts also mandate to ASDD (Australian Spatial Data Directory) deposit and this is now a thing of the past - replaced by http://find.ga.gov.au/ . I assume the ANZLIC standard is still in use. I am tending towards using DC spatial coverage to accomodate Geonames data (one instance for place name and another for coords). However, Kevin, you might need to investigate if this is the best solution??

**Comment by kevinpdavies** [comment]
I think it is important to distinguish between metadata describing a non-spatial (or aspatial) dataset (e.g. an interview transcript) that has a spatial attribute, and metadata describing a spatial dataset. The coverage attribute of DC seems like a nice, lightweight solution for storing a point attribute or rectangle extent (but not an irregular polygon unfortunately) with datum/projection. 

If we are storing spatial data (usual described as vector or raster), then the rich set of metadata described in the ANZLIC (or ISO19115, they are similar) is required. Spatial data requires technical attributes such as scale, resolution, scale factors (and lots more) in order for a user to correctly apply and interpret the data. So ideally, we use the DC metadata (or some other non-spatial standard) for non-spatial items, as well as support a richer standard (such as ISO19115) when adding a spatial dataset (it may have a parent DC record as well but that is implementation).

Regarding @ptsefton comment on RDFa or Microformats  for tagging text, this is out of my area of expertise, but happy to investigate.

**Comment by kevinpdavies** [comment]
The "ANZLIC ISO19115 profile" and ISO19115 are effectively the same. The major difference is a mandatory file identifier field in ANZLIC which is optional in ISO. This allows copies of the same record to be tracked, as well as parent/child relationships. ANZLIC is therefore preferred. ANZLIC also adheres to the ISO19139 XML implementation of ISO19115 which allows for validation against the schema.
Reference: http://www.anzlic.gov.au/__data/assets/pdf_file/0005/47318/ANZLICmetadataProfileGuidelines_v1-2.pdf
Would a subset of ANZLIC be suitable for non-spatial data in preference to DC? I will keep looking...

**Comment by kevinpdavies** [comment]
ANZLIC does support non-geographic data - see page 359 in profile guidelines (link in previous post) for an example. We could therefore use just ANZLIC. Correspondance with DC would be need to be checked, and support for harvesting of ANZLIC as DC added to Omeka. Now I am probably off topic...I'll stop now.

**Comment by sharynys** [comment]
No, not at all off topic - very little is, in this field (which is the best thing about it!). Thanks

**Comment by ptsefton** [comment]
@kevinpdavies I'll look into the RDFa stuff, and report back here. Not sure if we'd want to mess with Neatline, but we should be considering how to mark up semantics in documents / descriptions etc in the nicest possible way.

Here's an example [from the W3C](http://www.w3.org/MarkUp/2009/rdfa-for-html-authors):

```
<p about="[_:Amsterdam]"><span property="dc:name">Amsterdam</span> is located
   at latitude <span property="geo:lat" content="52.373">52°22'23"N</span>
 and longitude <span property="geo:long" content="4.892">4°53'32"E</span></p>
```

We could put stuff like this in transcripts that mention places, or species or whatever, then have it picked up by Solr indexing, and/or automatically appended to the item's metadata.

**Comment by Conal-Tuohy** [comment]
I agree with @ptsefton that if you're going to embed machine-readable semantics in your markup, then it should be done in an open and interoperable way. These Neatline spans with their HTML5 data-neatline-slug attributes don't cut it, really, because "data-neatline-slug" means nothing except to Neatline. NB if there's RDFa, instead, then it can be parsed and handled client side in Javascript, just as the Neatline spans are, but it can also be easily (and in a loosely coupled way) scraped and pushed into a SPARQL store for simple access to aggregated data.

**Comment by kevinpdavies** [comment]
Is JSON-LD also a viable alternative? The proposed GeoJSON-LD also has much richer geo properties 
http://geojson.org/vocab

**Comment by Conal-Tuohy** [comment]
JSON-LD can be embedded in markup, but it's not usually done. Such datasets are just floating islands of data, not connected to the text of the page. See http://www.w3.org/TR/json-ld/#embedding-json-ld-in-html-documents

But the RDF vocabulary defined by GeoJSON-LD (not the JSON syntax, but the ontological model itself) should be expressible in RDFa or any other RDF syntax.
